utils/pkb_automated_runner.sh

```diff
@@ -18,32 +18,36 @@ function run_pkb_provision() {
 function run_pkb_benchmark() {
     local file="$1"
     local unique_run_id="$2"
+    local subdir="$3"
     python "$PKB_PATH"/pkb.py --benchmarks=fio --run_uri="$unique_run_id" \
       --num_vms=1 --benchmark_config_file=pkb_fio_flags.yaml \
       --fio_jobfile="$file" --run_stage=run,cleanup,teardown
-    cp -r /tmp/perfkitbenchmarker/runs/"$unique_run_id" "$OUT_DIRECTORY"
+    if [[ ! -d "$OUT_DIRECTORY/$subdir" ]]; then
+        mkdir -p "$OUT_DIRECTORY/$subdir"
+    fi
+    cp -r /tmp/perfkitbenchmarker/runs/"$unique_run_id" "$OUT_DIRECTORY/$subdir"
 }
 
-
 if [ $# -lt 3 ]; then
   echo "Too few arguments supplied, should be in format [vms, output directory, config file]"
   exit -1
 fi
 
 if [[ ! -d "$OUT_DIRECTORY" ]]; then
-    mkdir -p "$OUT_DIRECTORY"
+  mkdir -p "$OUT_DIRECTORY"
 fi
 
 
 python fio_job_file_generator.py "$CONFIG_FILE"
 for file in generated_fio_files/*; do
   run_ids=
+  subdir="$(basename $file .fio)"
```

Comment:
I agree with your 2nd comment and have changed the variable name to be match the variable style.

comment 1 though, I disagree with.  If I move it below line 49 (within the for-loop) then we take a performance hit.  The file name doesn't change within the loop at 49 so the sub_directory variable would be updated to the same thing every loop.  I say better to avoid it and assign the value to sub_directory right below the for-loop at line 42.  Of course, maybe I am misunderstanding something about the code in this section (hope not, gulp).  If so, please let me know.